setuptools/monkey.py

```diff
@@ -152,13 +153,13 @@ def patch_for_msvc_specialized_compiler():
     Patch functions in distutils to use standalone Microsoft Visual C++
     compilers.
     """
-    # import late to avoid circular imports on Python < 3.5
-    msvc = import_module('setuptools.msvc')
-
-    if platform.system() != 'Windows':
+    if not get_platform().startswith('win'):
         # Compilers only availables on Microsoft Windows
         return
 
+    # import late to avoid circular imports on Python < 3.5
```

Comment:
Having this import appear before the Windows check means that when running tests on non-Windows, this module is still importable. Ideally, this import would appear at the top of the file except for the circular import issue (and will move to the top when Python 3.5 is the minimum version supported).

There is already a protection in `setuptools.msvc:48` to suppress errors importing msvc9compiler, but it's currently only catching an ImportError. Perhaps it should also catch the reported DistutilsPlatformError.

Presumably, this same issue would occur if someone were to install MSVC 6.0 or 7.0 (if that's even still possible).